view/ViewableData.php

```diff
@@ -127,6 +134,41 @@ public function __set($property, $value) {
 			$this->setField($property, $value);
 		}
 	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function __call($method, $arguments) {
+		$this->checkAndRedefineMethods();
+		return parent::__call($method, $arguments);
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function hasMethod($method) {
+		$this->checkAndRedefineMethods();
```

Comment:
Maybe this should be on setFailover instead, rather than duplicating it over all methods that rely on the failover?